src/project/python.rs

```diff
@@ -25,6 +26,22 @@ pub enum ParsePyPiRequirementError {
     MissingOperator(String),
 }
 
+impl fmt::Display for PyPiRequirement {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let all_extras = self
+            .extras
+            .clone()
```

Comment:
I think this clone is unnecessary, you can probably use `as_ref` instead.

Reply:
this bit was refactored